aten/src/ATen/native/cuda/Unique.cu

```diff
@@ -63,16 +64,24 @@ template <typename scalar_t>
         input_data, output_data, inverse_indices_data, num_inp, num_out);
     }
 
+    Tensor counts = at::empty({0}, self.options().dtype(kLong));
+    if (return_counts) {
+      counts.resize_(output.sizes()).fill_(0);
+      Tensor val = at::ones({num_inp}, self.options().dtype(kLong));
+      counts.scatter_add_(0, inverse_indices.view(-1), val);
```

Comment:
Humm, a broadcasting `scatter_add_` would have been very useful here :)

Reply:
Yeah, you mean such that we can just pass a single 1 as the value?
Is there a more efficient version of this code?